### PR TITLE
executable to get indexer reps and likelihood for 397 systematic python

### LINKED ISSUE #397: build infrastructure for systematic exploration of subsplit support in Python
* [x] take a well-sampled posterior
* [x] introduce in some out-of-sample NNIs
* [x] infer branch lengths using iqtree; I think we can do this [with the `-te` option](http://www.iqtree.org/doc/Command-Reference)
* [x] order the trees in terms of their iqtree likelihoods
* [x] write a variant of `HotStartBranchLengths` that takes the first branch length, not the mean
* [x] generate all trees in a subsplit DAG
* [x] evaluate their likelihoods
* [x] write out the trees, encoded as indicator matrices, along with their likelihoods to file(s) consumable by numpy
* [ ] find how much of the posterior support we get by greedy NNI search from an ML tree

### PR DESCRIPTION
**Pull request by chrisjenningsshaffer**
## Description
Moved noodle to extras folder. Added extras/reps_and_likelihoods with functionality to get relevant PCSP indexer representations and likelihoods needed for proof of concept nni sdag support. Modified gpinstance::TakeFirstBranchLengths.

Closes #397

## Tests
TakeFirstBranchLengths is tested in gp_doctest for the trees of data/hotstart.fasta

## Checklist:

* [x] Code follows our detailed [contribution guidelines](CONTRIBUTING.md)
* [x] `clang-format` has been run
* [x] TODOs have been eliminated from the code
* [x] Comments are up to date, document intent, and there are no commented-out code blocks
* [x] GitHub CI build on PR branch completed successfully

### CODE REVIEW

File: extras/reps_and_likelihoods.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// representations of various trees as part of a proof of concept for nni sdag support. 
+// Specifically, this takes in a fasta file and three newick files of rooted trees with
+// branch lengths. A subsplit DAG is built on the first set of trees and this is used
+// for the indexer representations of all three sets of trees. When a subsplit appears 
+// in the second or third set of trees, but not the subsplit DAG, the subsplit has 
+// SIZE_MAX as the index.
+
+#pragma once
+
+#include "../src/unrooted_sbn_instance.hpp"
```
Comment (matsen) [inline extras/reps_and_likelihoods.hpp:14]: @ognian- can you help us with these relative paths?

File: extras/reps_and_likelihoods.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  ur_inst.PrepareForPhyloLikelihood(simple_specification, thread_count, {}, true,
+                                    all_trees.TreeCount());
+  const auto log_likelihoods = ur_inst.UnrootedLogLikelihoods(all_trees);
+
+  WriteTreesToFile(out_path, all_representations, log_likelihoods);
+  WriteTreesToFile(credible_out_path, cred_representations);
+  WriteTreesToFile(pp_out_path, pp_representations);
+}
+
+void LoadIndexerRepresentations(
```
Comment (matsen) [inline extras/reps_and_likelihoods.cpp:55]: It seems like we could rewrite this to just return the indexer representation, which is stylistically preferable to mutating it in place.

**Review by matsen — CHANGES_REQUESTED** [review]

File: test/test_bito.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import numpy as np
 import bito
 import bito.beagle_flags as beagle_flags
+import os
```
[the remark this answers is not in the document]
Comment (chrisjenningsshaffer) [reply test/test_bito.py:12]: Nope. I think these changes are from a merge and aren't actually mine (the string.format isn't my usual style).

File: extras/reps_and_likelihoods.cpp (hunk 2 above)
Comment (chrisjenningsshaffer) [reply extras/reps_and_likelihoods.cpp]: I must have been going back and forth about pass by value and pass by reference with old school C++.

I've pushed a new version.

File: extras/reps_and_likelihoods.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  all_trees_gp_inst.MakeEngine();
+  all_trees_gp_inst.TakeFirstBranchLength();
+  cred_r_inst.ReadNewickFile(credible_rooted_nwk_path);
+  pp_r_inst.ReadNewickFile(pp_rooted_nwk_path);
+
+  auto all_trees = all_trees_gp_inst.GenerateCompleteRootedTreeCollection();
+  auto cred_trees = cred_r_inst.tree_collection_;
+  auto pp_trees = pp_r_inst.tree_collection_;
+  auto indexer = all_trees_gp_inst.GetDAG().BuildGPCSPIndexer();
+  //std::vector<RootedIndexerRepresentation> all_representations;
```
Comment (matsen) [inline extras/reps_and_likelihoods.cpp:39]: The new version looks great, though can you clear out these comments please?

**Review by matsen — CHANGES_REQUESTED** [review]

File: extras/reps_and_likelihoods.cpp (hunk 4 above)
Comment (chrisjenningsshaffer) [reply extras/reps_and_likelihoods.cpp]: Oops. Yes, fixed.

**Review by matsen — APPROVED** [review]

**Pull request merged**